src/sif/truckcost.cc

```diff
@@ -439,7 +439,8 @@ inline bool TruckCost::Allowed(const baldr::DirectedEdge* edge,
                                uint8_t& restriction_idx) const {
   // Check access, U-turn, and simple turn restriction.
   if (!IsAccessible(edge) || (!pred.deadend() && pred.opp_local_idx() == edge->localedgeidx()) ||
-      ((pred.restrictions() & (1 << edge->localedgeidx())) && !ignore_restrictions_) ||
+      ((pred.restrictions() & (1 << edge->localedgeidx())) &&
+       (!ignore_restrictions_ && !ignore_common_restrictions_)) ||
```

Comment:
- also needs to be added to the reverse equivalent
- also needs to be added to all other costings (pedestrian probably not)
- a bit nitty: but we can define a `ignore_turn_restrictions` somewhere in dynamiccost which will be set to `ignore_turn_restrictions = ignore_restrictions_ || ignore_common_restrictions_` so we can avoid the `&&` right here for every single call of `Allowed`